modules/translation/translation.go

```diff
@@ -108,6 +105,17 @@ func InitLocales() {
 	sort.Slice(allLangs, func(i, j int) bool {
 		return strings.ToLower(allLangs[i].Name) < strings.ToLower(allLangs[j].Name)
 	})
+
+	if !setting.IsProd {
+		watcher.CreateWatcher(ctx, "Locales", &watcher.CreateWatcherOpts{
+			PathsCallback: options.WalkLocales,
+			BetweenCallback: func() {
+				lock.Lock()
+				defer lock.Unlock()
```

Comment:
Can we always be certain that `lock == nil <==> setting.IsProd`?
Or is there even a remote possibility that this might be changed at some point?
Because if yes, we will have a NPE here…

Reply:
Look at the code.

Reply:
I don't mean **now**, I meant **in the future**.
I've seen the current code, and here it's a valid assumption.
However, I can't predict the future:
Some code hasn't been changed in a few years.
And other parts get changed on a seemingly monthly basis.
That's why I asked, because once someone changes this assumption, there's a possiblity he won't notice that he created an NPE with it as well…

Reply:
The assumption is made within the same function. 

Any situation that messes with the locking at line 52 that doesn't deal with the lock here and continues to allow the Watcher is a concurrency race and it deserves to NPE and bring down the development server until they think more carefully about concurrency.

It really is also worth remembering that this is ONLY used when `!setting.IsProd`

All of this code is for dev servers.